estargz/errorutil/errors.go

```diff
@@ -0,0 +1,51 @@
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package errorutil
+
+import (
+	"fmt"
+	"strings"
+)
+
+// Errors provides an abstraction for accumulating multiple errors.
+type Errors []error
+
+// Append adds an error to the Errors
+func (e *Errors) Append(err error) *Errors {
+	if e == nil {
+		return &Errors{err}
+	}
+	errs := append(*e, err)
+	return &errs
+}
+
+// Error implements error
+func (e *Errors) Error() string {
+	if e == nil {
+		return "0 errors occurred"
+	} else if len(*e) == 1 {
+		return (*e)[0].Error()
+	}
+	points := make([]string, len(*e)+1)
+	points[0] = fmt.Sprintf("%d error(s) occurred:", len(*e))
+	for i, err := range *e {
+		points[i+1] = fmt.Sprintf("* %s", err)
+	}
+	return strings.Join(points, "\n\t")
+}
+
+var _ error = (*Errors)(nil)
```

Comment:
We have tests, so we do not need this line

Reply:
I reworked the change pretty dramatically, so I'm no longer implementing a custom error type at all anyways.